File: app/src/main/java/net/squanchy/onboarding/account/AccountOnboardingInjector.java
package net.squanchy.onboarding.account;

import android.support.v7.app.AppCompatActivity;

import net.squanchy.injection.ActivityContextModule;
import net.squanchy.injection.ApplicationInjector;

final class AccountOnboardingInjector {

    private AccountOnboardingInjector() {
        // no instances
    }

    public static AccountOnboardingComponent obtain(AppCompatActivity activity) {
        return DaggerAccountOnboardingComponent.builder()
                .activityContextModule(new ActivityContextModule(activity))
                .applicationComponent(ApplicationInjector.obtain(activity))
                .build();
    }
}

File: app/src/main/java/net/squanchy/onboarding/account/AccountOnboardingInjector.kt
package net.squanchy.onboarding.account

import android.support.v7.app.AppCompatActivity

import net.squanchy.injection.ActivityContextModule
import net.squanchy.injection.ApplicationInjector

internal object AccountOnboardingInjector {

    fun obtain(activity: AppCompatActivity): AccountOnboardingComponent {
        return DaggerAccountOnboardingComponent.builder()
                .activityContextModule(ActivityContextModule(activity))
                .applicationComponent(ApplicationInjector.obtain(activity))
                .build()
    }
}
